lib/matplotlib/finance.py

```diff
@@ -816,6 +816,65 @@ def _candlestick(ax, quotes, width=0.2, colorup='k', colordown='r',
     return lines, patches
 
 
+def _check_input(opens, closes, highs, lows, miss=-1):
+    """Checks that *opens*, *highs*, *lows* and *closes* have the same length.
+    NOTE: this code assumes if any value open, high, low, close is
+    missing (*-1*) they all are missing
+
+    Parameters
+    ----------
+    ax : `Axes`
+        an Axes instance to plot to
+    opens : sequence
+        sequence of opening values
+    highs : sequence
+        sequence of high values
+    lows : sequence
+        sequence of low values
+    closes : sequence
+        sequence of closing values
+    miss : int
+        identifier of the missing data
+
+    Raises
+    ------
+    ValueError
+        if the input sequences don't have the same length
+    """
+
+    def _missing(sequence, miss=-1):
+        """Returns the index in *sequence* of the missing data, identified by
+        *miss*
+
+        Parameters
+        ----------
+        sequence :
+            sequence to evaluate
+        miss :
+            identifier of the missing data
+
+        Returns
+        -------
+        where_miss: numpy.ndarray
+            indices of the missing data
+        """
+        return np.where(np.array(sequence) == miss)[0]
+
+    same_length = (len(opens) == len(highs)) and (len(opens) == len(lows)) and\
+                  (len(opens) == len(closes))
+    _missopens = _missing(opens)
+    same_missing = (_missopens == _missing(highs)).all() and (_missopens ==
+        _missing(lows)).all() and (_missopens == _missing(closes)).all()
+
+    if not (same_length and same_missing):
+        msg = """
+        *opens*, *highs*, *lows* and *closes* must have the same length.
```

Comment:
This is going to result in a message with tabs and new lines.  I think this is better done as

```
msg = ("*opens*, *highs*, *lows* and *closes* must have the same length. " +
       "NOTE: this code assumes if any value open, high, low, close is " +
       "missing (*-1*) they all must be missing")
```